pkg/kubelet/prober/prober.go

```diff
@@ -165,31 +162,31 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 		return pb.exec.Probe(pb.newExecInContainer(container, containerID, command, timeout))
 	}
 	if p.HTTPGet != nil {
-		scheme := strings.ToLower(string(p.HTTPGet.Scheme))
-		host := p.HTTPGet.Host
-		if host == "" {
-			host = status.PodIP
-		}
-		port, err := extractPort(p.HTTPGet.Port, container)
+		req, err := httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
 		if err != nil {
 			return probe.Unknown, "", err
 		}
-		path := p.HTTPGet.Path
-		klog.V(4).InfoS("HTTP-Probe", "scheme", scheme, "host", host, "port", port, "path", path, "timeout", timeout)
-		url := formatURL(scheme, host, port, path)
-		headers := buildHeader(p.HTTPGet.HTTPHeaders)
-		klog.V(4).InfoS("HTTP-Probe Headers", "headers", headers)
+		if klogV4 := klog.V(4); klogV4.Enabled() {
+			port := req.URL.Port()
+			host := req.URL.Hostname()
+			path := req.URL.Path
+			scheme := req.URL.Scheme
+			klogV4.InfoS("HTTP-Probe", "scheme", scheme, "host", host, "port", port, "path", path, "timeout", timeout)
+			if headers := p.HTTPGet.HTTPHeaders; len(headers) > 0 {
+				klogV4.InfoS("HTTP-Probe Headers", "headers", headers)
+			}
```

Comment:
combine these 2 log lines. I think there's a risk of multiple probers interleaving their logs, in which event it wouldn't be clear which request the headers went with.

Reply:
I tried, but code quality checks complain, because they don't like using the slice with `...` appended to it; the code quality checks complain about an uneven number of keys and values.

Reply:
I suppose I could avoid the slice expansion that you had in https://github.com/tallclair/kubernetes/commit/8fd8a3d823198e92a49347ae495900aa0cea09aa#diff-6c7396231fee734cca8c9195e23b9ff3893577c3046c61268359d3bec25d00ccR174-R176 with an if/else for the `InfoS` call.

e.g.
```
if headers := p.HTTPGet.HTTPHeaders; len(headers) == 0 {
    klogV4.InfoS("HTTP-Probe", "scheme", scheme, "host", host, "port", port, "path", path, "timeout", timeout)
else {
    klogV4.InfoS("HTTP-Probe", "scheme", scheme, "host", host, "port", port, "path", path, "timeout", timeout, "headers, headers)
}
```

Would you prefer that or would your prefer that to leave this as-is?

Reply:
I'd just go with:
```
    klogV4.InfoS("HTTP-Probe", "scheme", scheme, "host", host, "port", port, "path", path, "timeout", timeout, "headers, headers)
```

IMO it's fine to include the empty headers in the log line.